### ISSUE TITLE
Feature request: simple install directory choice

### ISSUE THREAD

**Issue by henryiii** [TACC/Lmod#352]
LMod currently forces its install structure on the user; for example, if you set prefix to `$prefix/lmod/VERSION/`, LMod will install to `$prefix/lmod/VERSION/lmod/VERSION/`. For systems (like Brew) that already manage versions, it would be much better to not force this structure on the user. What's worse, when Brew makes a link to `/usr/opt/lmod`, the path to the init script, for example, becomes: `/usr/opt/lmod/lmod/lmod/init/sh`. (In https://github.com/Homebrew/homebrew-core/pull/24305, I've removed one level of `lmod` by installing to `$prefix/lmod/VERSION/versions/lmod/VERSION/` and then making a symlink, but that means you are following two symlinks to an even longer final path.)

The request is to add a `./configure` option to allow direct specification the final install dir with no redirect. 

Ideally, if there are no collisions in /libexec, you could even install luamodules and lmod into the same directory, causing the path to be `/usr/opt/lmod/init/sh` if specifying the actual installation directory was possible.

**Comment by rtmclay** [comment]
Well Lmod does this for a reason.  The reason is that a site (or brew user) needs to source the init/profile or init/sh script during startup.  It would be a bad idea to have to change that every time there is a new version of Lmod installed.   The only interface that matters with Lmod is the module and ml shell functions.  

It seems that this is a cosmetic rather than technical issue.

**Comment by henryiii** [comment]
At this point it's mostly cosmetic but something that should be eventually fixed, IMO. The problem is that a system like brew already has versioning built in, and a symlink system for all packages built in too (since it's not a unique problem to LMod). (Packages live in `/usr/local/Cellar/package/version`, and the latest version is symlinked to `/usr/opt/package`)

The original solution in Brew was this:
```ruby
system "./configure", "--prefix=#{prefix}"
system "make", "install"
# Move prefix/lmod/VERSION/lmod/VERSION/* to prefix/lmod/VERSION/*
mv Dir[prefix/"lmod/#{version}/*"], prefix
rmdir prefix/"lmod/#{version}"
(prefix/"lmod").install_symlink ".." => version
Dir[prefix/"lmod/init/*"].reject { |f| f["lmodrc.lua"] }.each do |f|
    inreplace f, "lmod/lmod", "lmod"
end
```

which I've been able to reduce to this:

```ruby
system "./configure", "--prefix=#{prefix}/versioned"
system "make", "install"
prefix.install_symlink "versioned/lmod/lmod" => "lmod"
```

which is not ideal but not too bad (IMO).

**Comment by henryiii** [comment]
PS: "Fixed" == "user choice"; having a versioned install or even a default versioned install is not the problem, just not being able to deactivate it if a user wants to.

**Comment by rtmclay** [comment]
I have release 7.7.21 which has a config option called --with-siteControlPrefix=yes.  This means that --prefix=<path> will be where Lmod is installed not <path>/lmod/<version>. Both the configure and make install steps remind installers that the Lmod scripts in /etc/profile.d must be updated.

**Comment by rtmclay** [comment]
I missed one of the Make Variable that controlled  the location of where the term package went.  I have now released 7.7.22 which should fix this.

**Comment by henryiii** [comment]
Thanks! LMod in Homebrew is much nicer now.
